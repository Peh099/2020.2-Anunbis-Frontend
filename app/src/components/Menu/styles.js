import styled from 'styled-components';
import Button from '../Button';

export const MenuBar = styled.div`
    background-color: #212121;
    display: flex;
    height: 40px;
`

export const Logo = styled.div`
    width: 280px;
`

export const ImageLogo = styled.img`
        max-width: 190px;
        display: inline-block;
`

export const ProfessorSearchStyle = styled.div`
    display: flex;
    align-items: center;
    height: 40px;
    width: 400px;
    Input {
        color: #FFFDE7;
        border-bottom: 1px solid #FFFDE7;
        margin: 0px;   
    }
    input {
            height: 18px;
    }
    input::-webkit-input-placeholder  {
        height: 30px;
        color: rgba(255, 255, 255, 0.603);
    }
`

export const BtnEdition = styled (Button)`
    margin-top: 10px;
    letter-spacing: 1.5px;
    color: #FFD54F;
    border: none;
    box-shadow: none;
`
export const Container = styled.div`
    background-color: inherit;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100px;
    height: 2px;
    margin-left: auto;
    margin-inline-end: 50px;
    letter-spacing: 1.5px;
    color: #FFD54F;
    box-shadow: none;
`
