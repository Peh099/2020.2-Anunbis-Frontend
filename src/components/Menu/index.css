.Menu {
    width: 100vw;
    height: 10vh;
    background-color: #212121;
}