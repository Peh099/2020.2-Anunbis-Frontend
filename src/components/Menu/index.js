import React from 'react';
import './index.css'
import Name_Logo from "../../assets/Name_Logo.png";
import Input from '../Input';
import {useHistory} from "react-router-dom";
import styled from 'styled-components';

const ProfessorSearch = () =>{
    const history =  useHistory()

    const onSubmit = (data) => {
        if(data.key === 'Enter' && data.target.value.trim().length > 0)
           history.push("/professor/search/" + data.target.value.trim()) 
    }

    return (<ProfessorSearchStyle><Input type="text" width="400px" text="Informe o nome do professor" onkeydown={onSubmit}/></ProfessorSearchStyle>)
}


export default function Menu() {
    return (
        <MenuBar>
            <Logo>
                <ImageLogo src={Name_Logo} alt="logo"></ImageLogo>
            </Logo>
            <ProfessorSearch />
        </MenuBar>
    );
}


const MenuBar = styled.div`
    background-color: #212121;
    display: flex;
    height: 40px;
    position: absolute;
    width: 100vw;
`;

const Logo = styled.div`
    width: 280px;
`;

const ImageLogo = styled.img`
        max-width: 190px;
        display: inline-block;
`;

const ProfessorSearchStyle = styled.div`
    display: inline-block;
    height: 40px;
    width: 400px;
    Input {
        
        color: #FFFDE7;
        border-bottom: 1px solid #FFFDE7;
    }
    input::-webkit-input-placeholder  {
        height: 30px;
    color: rgba(255, 255, 255, 0.603);
}
`;