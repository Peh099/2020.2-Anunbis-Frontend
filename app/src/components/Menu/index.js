import React from 'react';
import Name_Logo from "../../assets/images/Name_Logo.png";
import Input from '../Input';
import {useHistory} from "react-router-dom";
import Button from '../Button';
import MenuOptions from '../MenuOptions';
import {logOut} from '../../services/Auth';
import { MenuBar, Logo, ImageLogo, ProfessorSearchStyle, BtnEdition, Container } from './styles.js';

function ProfessorSearch ({history}) {
    const onSubmit = (data) => {
        if(data.key === 'Enter' && data.target.value.trim().length > 0)
            history.push("/professor/search/" + data.target.value.trim())
    }
    return (<ProfessorSearchStyle><Input type="text" width="400px" text="Informe o nome do professor" onkeydown={onSubmit}/></ProfessorSearchStyle>)
} 

export default function Menu() {

    const history = useHistory();
    const [menuOptions, setMenuOptions] = React.useState("");

    function makeMenuOptions(){
        if(menuOptions===""){
            return (setMenuOptions(
                <MenuOptions>
                    <Button type='button' backColor='#FFD54F' text='CONFIGURAR' padding='3px' onClick={() => {setMenuOptions(""); history.push('/profile')}}/>
                    <Button backColor='#FFD54F' text='SOBRE' padding='3px' onClick={() => console.log("sobre")} />
                    <Button backColor='#FFD54F' text='SAIR' padding='3px' onClick={() => {logOut(); history.push('/user/login')}} />
                </MenuOptions>
            ))
        }else{
            setMenuOptions("");
        }
    }

    return (
        <MenuBar>
            <Logo>
                <ImageLogo src={Name_Logo} alt="logo"></ImageLogo>
            </Logo>
            <ProfessorSearch history={history}/>
            <Container>
                <BtnEdition text='ººº' padding='5px 5px' backColor='#212121' onClick={() => makeMenuOptions()} />
                {menuOptions}
            </Container>
        </MenuBar>
    );
}