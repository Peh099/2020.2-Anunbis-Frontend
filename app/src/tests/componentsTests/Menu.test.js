import renderer from 'react-test-renderer';
import Menu from '../../components/Menu';

describe('Snapshot Menu component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Menu />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})