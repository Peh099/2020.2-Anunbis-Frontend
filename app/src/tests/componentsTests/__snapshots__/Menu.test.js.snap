// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Menu component matches the snapshot 1`] = `
<div
  className="sc-bdnxRM jssPjp"
>
  <div
    className="sc-gtsrHT iVoEWf"
  >
    <img
      alt="logo"
      className="sc-dlnjwi fOZsPb"
      onClick={[Function]}
      src="Name_Logo.png"
    />
  </div>
</div>
`;
